dap_py/dap.py

```diff
@@ -0,0 +1,610 @@
+import os
+import json
+import base64
+from pathlib import Path
+import requests
+from getpass import getpass
+from .utils.scraper import get_api_url
+
+class BadStatusCodeError(RuntimeError):
+    def __init__(self, req):
+        self.status_code = req.status_code
+        try:
+            self.reason = req.json()['message']
+        except (KeyError, ValueError):
+            self.reason = req.reason
+
+    def __str__(self):
+        return ((
+            'Server Returned Bad Status Code\n'
+            f'Status Code: {self.status_code}\n'
+            f'Reason: {self.reason}')
+        )
+
+
+class dap:
```

Comment:
I think class names should begin with uppercase character per convention.  I think `DAP` is appropriate.